torch/distributed/_tensor/random.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+from typing import Optional
+
+import torch
+from torch import Tensor
+
+from torch.distributed._tensor import DeviceMesh
+
+def set_rng_state(new_state: Tensor, device_mesh: DeviceMesh) -> None:
+    r"""
+    """
+    if device_mesh is not None and device_mesh.get_coordinate() is not None:
+        # the current rank is in mesh
+        if device_mesh.device_type == "cuda":
+            torch.cuda.set_rng_state(new_state)
+        else:
+            raise NotImplementedError(
+                f"DTensor randomness only supports cuda device type, "
+                f"but got {device_mesh.device_type}"
+            )
+
+def get_rng_state(device_mesh: DeviceMesh) -> Optional[Tensor]:
+    r"""
+    """
+    if device_mesh is None or device_mesh.get_coordinate() is None:
```

Comment:
I think we should always give an rng state when user call rng state? i.e. we can return the cuda rng as long as the device_mesh type is correct. 

Also, we should probably assert device_mesh must be passed in, instead of silently returning None to user.

Reply:
right, for SPMD we should not return None for non-participating ranks.